### DISCUSSION TITLE
Could you recommend which dotnet approach is better to work with OpenAI?

### DISCUSSION THREAD

**Discussion by YuriyMorozyuk95** [microsoft/teams-sdk#189] [Q&A]
- Semantic Ketnel - https://github.com/microsoft/semantic-kernel/blob/main/dotnet/README.md
- Azure Open AI SDK - https://learn.microsoft.com/en-us/dotnet/api/overview/azure/ai.openai-readme?view=azure-dotnet-preview
- https://github.com/betalgo/openai ?
- https://github.com/OkGoDoIt/OpenAI-API-dotnet ?
- https://github.com/RageAgainstThePixel/OpenAI-DotNet ?

Or your solution?

Do there is any comparision?

**Comment by singhk97** [comment]
This is how I would go about choosing which repo:

1. If I want to only use OpenAI's text/chat completion API, and possibly AzureOpenAI's: `Azure OpenAI SDK`
2. If I want to access other OpenAI APIs like Embeddings, Moderation, DALL-E: Go with the community developed SDKs you mentioned
3. If I want to build complex & robust LLM AI applications: `Semantic Kernel`
4. If I want to build conversation AI bots in Teams (and possibly other channels): use our solution
